### PR TITLE
supportsBatch shortcut

### LINKED ISSUE #43: supportsBatch with apollo-server
We use transport-level batching, yet we also use `apollo-server-express` so I don't want apollo-engine to split my batched requests back up again.

The docs of the engine say that you have to set `origins.supportsBatch` if  you want the full batched request to be passed through, but the issue is that when you use the `apollo-engine` package the `origins` option is automatically generated for you.

So how can we set `supportsBatch` to false when using `apollo-server-express` and `apollo-engine`?

### PR DESCRIPTION
**Pull request by thepwagner**
Previously this flag required defining an origin (and ignoring
TypeScript types!) - accept `supportsBatch: boolean` option to avoid
making users dive into origin config.

Fixes #43

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -30,6 +30,7 @@ export interface EngineConfig {
             url: string,
             headerSecret: string,
         }
+        supportsBatch?: boolean ,
```
Comment (glasser) [inline src/index.ts]: Weird space before comma on this and the following line (time for prettier?)

File: src/index.ts [hunk 2]
```diff
@@ -169,15 +173,17 @@ export class Engine {
...
+                        supportsBatch: this.supportsBatch,
                     }];
                 } else {
                     // Extend any existing HTTP origins with the chosen PSK:
                     childConfig.origins.forEach(origin => {
                         if (typeof origin.http === 'object') {
                             origin.http.headerSecret = this.middlewareParams.psk;
                         }
-                    })
+                        // Don't update `supportsBatch`
```
Comment (glasser) [inline src/index.ts]: Seems like this could be confusing. Maybe just don't override an explicitly configured supportsBatch?

**Comment by thepwagner** [comment]
Comments addressed.
Documentation could go there; I don't think writing a new documentation section should block this fix?

**Comment by thepwagner** [comment]
https://github.com/apollographql/apollo-engine-js/pull/58 for new documentation.

**Pull request merged**